airflow/providers/google/cloud/sensors/dataproc.py

```diff
@@ -0,0 +1,83 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains a Dataproc Job sensor.
+"""
+from typing import Optional
+
+from airflow.providers.google.cloud.hooks.dataproc import DataprocHook
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+from google.cloud.dataproc_v1beta2.types import JobStatus
+
+
+class DataprocJobSensor(BaseSensorOperator):
+    """
+    Check for the state of a job submitted to a Dataproc cluster.
```

Comment:
nit: you can use to dataproc API to submit a spark job to GKE ([beta docs](https://cloud.google.com/dataproc/docs/concepts/jobs/dataproc-gke)) therefore IMO it is more accurate to say
```suggestion
    Check for the state of a previously submitted Dataproc job.
```